### PR TITLE
Add $(TargetFramework) to workaround OutputPath

### PR DESCRIPTION
**Pull request by brthor**
@livarcocc @nguerrera @srivatsn

### CODE REVIEW

**Comment by livarcocc** [comment]
What is this workaround? Do we have an issue for the original problem somewhere? Should we also update the MSBuildTestApp?

File: src/dotnet/commands/dotnet-new/CSharp_MSBuild/$projectName$.csproj [hunk 1]
```diff
@@ -4,7 +4,7 @@
   <PropertyGroup>
     <OutputType>Exe</OutputType>
     <TargetFrameworks>netcoreapp1.0</TargetFrameworks>
-    <OutputPath>bin\$(Configuration)</OutputPath>
+    <OutputPath>bin\$(Configuration)\$(TargetFramework)</OutputPath>
```
Comment (eerhardt) [inline src/dotnet/commands/dotnet-new/CSharp_MSBuild/$projectName$.csproj]: This should already be happening in the SDK.  See

https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NETCore.Build.Tasks/build/netstandard1.0/Microsoft.NETCore.TargetFrameworkInference.targets#L86

Why do we need to do it here?

Comment (livarcocc) [inline src/dotnet/commands/dotnet-new/CSharp_MSBuild/$projectName$.csproj]: That's what I told @brthor at Slack.

**Comment by brthor** [comment]
> What is this workaround? Do we have an issue for the original problem somewhere? Should we also update the MSBuildTestApp?

The output path being present in the dotnet new template itself is a workaround for this issue: https://github.com/dotnet/cli/issues/4287

I'm simply adjusting the in place workaround to be consistent with the default in the sdk so migrated project references continue to work. 

I think we should update the test app as well.

File: src/dotnet/commands/dotnet-new/CSharp_MSBuild/$projectName$.csproj (hunk 1 above)
Comment (nguerrera) [inline src/dotnet/commands/dotnet-new/CSharp_MSBuild/$projectName$.csproj]: +1 to what @eerhardt and @livarcocc are saying. This doesn't seem like the right fix. I would expect you to get two levels of targetframework with this.

**Pull request closed without merging**